lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,33 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds qw(ONE_HOUR);
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => ONE_HOUR / 2;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache') . '/gitlab_maintenance.lock');
+}
+
+sub remove_state_file ($app) { _state_file($app)->remove }
+
+sub decide_outcome ($app) {
+    my $file = _state_file($app);
+    unless (-f $file) {
+        $file->touch;
+        return SKIP;
+    }
+    my $file_age = time - _state_file($app)->stat->mtime;
```

Comment:
This repeats `_state_file($app)` instead of reusing `$file`;

But also, there is a race condition.
If the file was removed between the `-f` check and this line, it will die with `Can't call method "mtime" on an undefined value`.

To fix this, you could just use `my $stat = _state_file($app)->stat` and check if that is defined or not for deciding whether the file exists; and then just call mtime on it